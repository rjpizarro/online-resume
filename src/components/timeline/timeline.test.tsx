import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent, waitFor } from '@testing-library/react'

jest.mock("@material-ui/core/Fade", () => (props) => <div>{props.children}</div>)

import Timeline from "./index"

const jobs = [
  {
    startDate: new Date("01/03/2014"),
    endDate: new Date("01/05/2015"),
    company:"Test company",
    jobTitle:"Test job title",
    description: "Job test description",
    id: 1
  },
  {
    startDate: new Date("01/10/2015"),
    endDate: new Date("01/11/2016"),
    company:"Test company 2",
    jobTitle:"Test job title 2",
    description: "Job test description 2",
    id: 2
  },
  {
    startDate: new Date("01/02/2017"),
    endDate: new Date("01/03/2018"),
    company:"Test company 3",
    jobTitle:"Test job title 3",
    description: "Job test description 3",
    id: 3
  },
]

const extraJob = {
  startDate: new Date("01/04/2018"),
  endDate: new Date("01/06/2019"),
  company:"Test company 4",
  jobTitle:"Test job title 4",
  description: "Job test description 4",
  id: 4
}

describe("Timeline", () => {
  it("render normally without jobs", () => {
    const tree = renderer
      .create(<Timeline />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("render normally with 3 jobs", () => {
    const tree = renderer
      .create(<Timeline jobs={jobs} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("render normally with more than 3 jobs", () => {
    const tree = renderer
      .create(<Timeline jobs={[...jobs, extraJob]} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("render normally with more than 3 jobs", () => {
    const tree = renderer
      .create(<Timeline jobs={[...jobs, extraJob]} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("show more jobs when click on View More", async () => {
    const { getByTestId, queryAllByTestId, rerender } = render(<Timeline jobs={[...jobs, extraJob]} />)

    const showMoreButton = getByTestId("show-more-button")
    expect(queryAllByTestId("job-description")).toHaveLength(3)
    expect(showMoreButton.innerHTML).toContain("Show More!")
    expect(showMoreButton).not.toBeDisabled()

    fireEvent.click(showMoreButton)

    await waitFor(() => {
      expect(queryAllByTestId("job-description")).toHaveLength(4)
      expect(showMoreButton.innerHTML).toContain("That's all folks! üê∑")
      expect(showMoreButton).toBeDisabled()
    })
  })
})