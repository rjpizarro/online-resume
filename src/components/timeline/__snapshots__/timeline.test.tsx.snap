// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline render normally with 3 jobs 1`] = `
<div
  className="makeStyles-timelineContainer-1"
>
  <div
    className="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  >
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2014 Jan
                   - 
                  2015 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2015 Jan
                   - 
                  2016 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company 2
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title 2
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description 2
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2017 Jan
                   - 
                  2018 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company 3
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title 3
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description 3
          </p>
        </div>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
    data-testid="show-more-button"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      That's all folks! üê∑
    </span>
  </button>
</div>
`;

exports[`Timeline render normally with more than 3 jobs 1`] = `
<div
  className="makeStyles-timelineContainer-1"
>
  <div
    className="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  >
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2014 Jan
                   - 
                  2015 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2015 Jan
                   - 
                  2016 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company 2
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title 2
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description 2
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2017 Jan
                   - 
                  2018 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company 3
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title 3
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description 3
          </p>
        </div>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    data-testid="show-more-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Show More!
    </span>
  </button>
</div>
`;

exports[`Timeline render normally with more than 3 jobs 2`] = `
<div
  className="makeStyles-timelineContainer-1"
>
  <div
    className="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  >
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2014 Jan
                   - 
                  2015 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2015 Jan
                   - 
                  2016 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company 2
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title 2
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description 2
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-completed"
    >
      <span
        className="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div>
      <div
        className="MuiStep-root"
        data-testid="job-description"
      >
        <span
          className="MuiStepLabel-root MuiStepLabel-horizontal"
        >
          <span
            className="MuiStepLabel-labelContainer"
          >
            <span
              className="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
            >
              <div>
                <span
                  className="MuiTypography-root MuiTypography-caption"
                  data-testid="dates"
                >
                  2017 Jan
                   - 
                  2018 Jan
                </span>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle1"
                >
                  Test company 3
                </h6>
                <h6
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  Test job title 3
                </h6>
              </div>
            </span>
          </span>
        </span>
        <div
          active={false}
          className="makeStyles-stepContent-2"
          completed={false}
          disabled={false}
          expanded={false}
          icon={NaN}
        >
          <p
            className="MuiTypography-root makeStyles-jobDescription-3 MuiTypography-body1"
          >
            Job test description 3
          </p>
        </div>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    data-testid="show-more-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Show More!
    </span>
  </button>
</div>
`;

exports[`Timeline render normally without jobs 1`] = `
<div
  className="makeStyles-timelineContainer-1"
>
  <div
    className="MuiPaper-root MuiStepper-root MuiStepper-vertical MuiPaper-elevation0"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
    data-testid="show-more-button"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      That's all folks! üê∑
    </span>
  </button>
</div>
`;
